pkg/prometheus/prometheus.go

```diff
@@ -0,0 +1,80 @@
+package prometheus
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const prometheusDefaultPort = "2112"
+
+var (
+	ApiCallsFailure = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "api_call_failures_total",
+		Help: "The total number of api calls that failed by the updater",
+	}, []string{"node", "errorType"})
+
+	OperationDelay = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "operation_delay",
+			Help: "Represent the latency of the update operation",
+		}, []string{"node"})
+)
+
+func extractErrorType(error string) string {
+	// TODO temporary until PodresourceAPI errors will be standardize
+	if strings.Contains(error, "GetAllocatableResources disabled") {
+		return "feature-gate"
+	}
+	return "unexpected"
+}
+
+func nodeName() string {
+	val, ok := os.LookupEnv("NODE_NAME")
+	if !ok {
+		return ""
```

Comment:
we need a fallback here. Let's use `https://pkg.go.dev/os#Hostname`. Even better: get the hostname as commandline flag and pass it through. Or get once in `InitPrometheus`

Reply:
to clarify: the key point is to get the hostname once and pass it through. Taking it from the env var is just fine, actually using a commandline arg is slightly worse option. We need the fallback using `os.Hostname`, though.